### PR TITLE
feat(influxql): CUMULATIVE_SUM window function

### LINKED ISSUE #7467: InfluxQL-IOx: Teach IOx the `CUMULATIVE_SUM` function
See [CUMULATIVE_SUM](https://docs.influxdata.com/influxdb/v1.8/query_language/functions/#cumulative_sum)

### PR DESCRIPTION
**Pull request by mhilton**
Closes #7467

Implement the InfluxQL CUMULATIVE_SUM window function. This is implemented as described in
https://docs.influxdata.com/influxdb/v1.8/query_language/functions/#cumulative_sum.

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by alamb — APPROVED** [review]

File: influxdb_iox/tests/query_tests/cases/in/window_like.influxql
Comment (stuartcarnie) [inline influxdb_iox/tests/query_tests/cases/in/window_like.influxql:1]: 👍🏻 test coverage looks great

File: iox_query_influxql/src/window/cumulative_sum.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/// PartitionEvaluator which returns the cumulative sum of the input.
+#[derive(Debug)]
+struct CumulativeSumPartitionEvaluator {}
+
+impl PartitionEvaluator for CumulativeSumPartitionEvaluator {
+    fn evaluate_all(&mut self, values: &[ArrayRef], num_rows: usize) -> Result<Arc<dyn Array>> {
+        assert_eq!(values.len(), 1);
+
+        let array = Arc::clone(&values[0]);
```
[the remark this answers is not in the document]
Comment (stuartcarnie) [reply iox_query_influxql/src/window/cumulative_sum.rs:41]: The difference is that InfluxQL drops intervals with empty values, unless joined with other rows with non-null values at the same timestamp. I ran into the same issue implementing `DIFFERENCE`, which could use `LAG`, except that `NULL` values are not ignored.

I updated the planner to show it using the built-in aggregate, `SUM`, and we can see where the `NULL` intervals appear, the InfluxQL implementation drops those values. 

CleanShot 2023-07-18 at 11 30 08@2x

@mhilton I added #8253 to add a test that combines a sparse column, showing expected `NULL` values. @alamb this test demonstrates where the built-in `SUM` would also return results inconsistent with InfluxQL behaviour.

Incidentally, Snowflake has an extended syntax, `IGNORE NULLS`:

```sql
SELECT usage_system - LAG(usage_system) IGNORE NULLS OVER ...
```

which I believe would be the same as InfluxQL's `DIFFERENCE`.

**Review by stuartcarnie — APPROVED** [review]
Looks great, @mhilton – I added #8253 to include a test that shows the expected results when rows contain `NULL` values when the projection contains other columns with non-`NULL` values.
